docs/dev/design/new-search-api.rst

```diff
@@ -0,0 +1,238 @@
+             }
+           },
+           {
+             "type": "domain",
+             "role": "http:post",
+             "name": "/api/v3/projects/",
+             "id": "post--api-v3-projects-",
+             "content": "Import a project under authenticated user. Example request: BashPython$ curl \\ -X POST \\ -H \"Authorization: Token <token>\" https://readthedocs.org/api/v3/projects/ \\ -H \"Content-Type: application/json\" \\ -d @body.json import requests import json URL = 'https://readthedocs.org/api/v3/projects/' TOKEN = '<token>' HEADERS = {'Authorization': f'token {TOKEN}'} data = json.load(open('body.json', 'rb')) response = requests.post( URL, json=data, headers=HEADERS, ) print(response.json()) The content of body.json is like, { \"name\": \"Test Project\", \"repository\": { \"url\": \"https://github.com/readthedocs/template\", \"type\": \"git\" }, \"homepage\": \"http://template.readthedocs.io/\", \"programming_language\": \"py\", \"language\": \"es\" } Example response: See Project details Note Read the Docs for Business, also accepts",
+             "highlights": {
+               "name": [],
+               "content": [
+                 ", json=data, headers=HEADERS, ) print(response.json()) The content of body.json is like,  &quot;name&quot;: &quot;<span>Test</span>"
+               ]
+             }
+           }
+         ]
+       }
+     ]
+   }
+
+Examples
+--------
+
+- ``project:docs project:dev/latest test``: search for ``test`` in the default
+  version of the ``docs`` project, and in the latest version of the ``dev`` project.
+- ``a project:docs/stable search term``: search for ``a search term`` in the
+  stable version of the ``docs`` project.
+
+- ``project:docs project\:project/version``: search for ``project::project/version`` in the
+  default version of the ``docs`` project.
+
+- ``search``: invalid, at least one project is required.
+
+
+Future features
+---------------
+
+- Allow searching on several versions of the same project
+  (the API response is prepared to support this).
+- Allow specify the type of search:
+
+  - Multi match (query as is)
+  - Simple query string (allows using the ES query syntax)
+  - Fuzzy search (same as multi match, but with with fuzziness)
+
+Questions / pending decisions
+-----------------------------
+
+Integration with the dashboard search.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The search API and the search from the dashboard
+use the same backend, but they are used in a different way.
+
+The dashboard search by default searches on all projects (.org),
+or all project the current user has access to (.com).
+And the API search searches on explicitly given projects.
+
+The dashboard search allows filtering by version and role,
+the API search allows filtering only by version (and it's required).
+
+The dashboard search makes use of filters in order to return
+the number of results from other versions/roles.
+Is this feature useful? It could slow down the response.
```

Comment:
Faceting is useful, but not sure how we'd represent it in the UI in this new search UX. I think if we lose it, it's not a huge deal.